src/tasks/netSdk/NetSdkTaskHelper.ts

```diff
@@ -0,0 +1,138 @@
+    | string;
+
+export const NET_SDK_RUN_TASK_SYMBOL = 'dotnet-container-sdk';
+const IMAGE_TAG = 'dev'; // intentionally default to dev tag for phase 1 of this feature
+const ERR_MSG_NO_WORKSPACE_FOLDER = l10n.t(`Unable to determine task scope to execute task ${NET_SDK_RUN_TASK_SYMBOL}. Please open a workspace folder.`);
+
+export class NetSdkTaskHelper {
+
+    public async getNetSdkBuildCommand(context: DockerTaskExecutionContext): Promise<string> {
+
+        const configuration = 'Debug'; // intentionally default to Debug configuration for phase 1 of this feature
+        const projPath = await this.inferProjPath(context.actionContext, context.folder);
+
+        // {@link https://github.com/dotnet/sdk-container-builds/issues/141} this could change in the future
+        const publishFlag = NetCoreTaskHelper.isWebApp(projPath) ? '-p:PublishProfile=DefaultContainer' : '/t:PublishContainer';
+
+        const folderName = await quickPickWorkspaceFolder(context.actionContext, ERR_MSG_NO_WORKSPACE_FOLDER);
+
+        const args = composeArgs(
+            withArg('dotnet', 'publish'),
+            withNamedArg('--os', await this.normalizeOsToRid()),
+            withNamedArg('--arch', await this.normalizeArchitectureToRid()),
+            withArg(publishFlag),
+            withNamedArg('--configuration', configuration),
+            withNamedArg('-p:ContainerImageName', getValidImageName(folderName.name), { assignValue: true }),
+            withNamedArg('-p:ContainerImageTag', IMAGE_TAG, { assignValue: true })
+        )();
+
+        const quotedArgs = Shell.getShellOrDefault().quote(args);
+        return quotedArgs.join(' ');
+    }
+
+    public async getNetSdkRunCommand(context: DockerTaskExecutionContext): Promise<string> {
+        const client = await ext.runtimeManager.getClient();
+        const folderName = await quickPickWorkspaceFolder(context.actionContext, ERR_MSG_NO_WORKSPACE_FOLDER);
+
+        const command = await client.runContainer({
+            detached: true,
+            publishAllPorts: true,
+            name: getDefaultContainerName(folderName.name, IMAGE_TAG),
+            environmentVariables: {},
+            removeOnExit: true,
+            imageRef: getDefaultImageName(folderName.name, IMAGE_TAG),
+            labels: defaultVsCodeLabels,
+            mounts: await this.getRemoteDebuggerMount(),
+            customOptions: '--expose 8080',
+            entrypoint: '/bin/sh'
+        });
+
+        const quotedArgs = Shell.getShellOrDefault().quote(command.args);
+        const commandLine = [client.commandName, ...quotedArgs].join(' ');
+
+        return commandLine;
+    }
+
+    public async inferProjPath(context: IActionContext, folder: WorkspaceFolder): Promise<string> {
+        const noProjectFileErrMessage = l10n.t('No .csproj file could be found.');
+        const item = await quickPickProjectFileItem(context, undefined, folder, noProjectFileErrMessage);
+        return item.absoluteFilePath || '';
```

Comment:
Why the `|| ''`? Is that to convince TypeScript that there's definitely a value being returned?

Reply:
my intention was that if `item.absoluteFilePath` is `undefined` then at least it'll return a empty string rather than throw an error? but after looking at `quickPickProjectFileItem` closer, it handles the case where a `fileItem` is not found so I'll remove the `|| ''`. good catch!